libsplinter/src/service/scabbard/rest_api.rs

```diff
@@ -272,27 +269,23 @@ pub fn make_get_batch_status_endpoint() -> ServiceEndpoint {
                 })
                 .map(Duration::from_secs);
 
-            let timeout = match wait {
-                Some(wait_time) => match Instant::now().checked_add(wait_time) {
-                    Some(t) => Some(t),
-                    None => {
-                        error!("Failed to determine wait time");
-                        return Box::new(
+            match scabbard.get_batch_info(ids, wait) {
+                Ok(iter) => match iter.collect::<Result<Vec<_>, _>>() {
+                    Ok(batch_infos) => Box::new(HttpResponse::Ok().json(batch_infos).into_future()),
+                    Err(err) => {
+                        error!(
+                            "Failed to get completed batch statuses before timeout: {}",
+                            err
+                        );
+                        Box::new(
                             HttpResponse::InternalServerError()
```

Comment:
I don't think this is an InternalServerError, but I couldn't find an error code that seemed to fit this situation.

Reply:
It could be `408: Request Timeout`

Reply:
Updated to 408